# Procurement KPI Analysis

## Introduction
This dataset contains procurement and supplier performance data covering
777 purchase orders across 13 columns. The analysis explores supplier
defective units, negotiation savings, delivery performance and compliance
to identify the most and least reliable suppliers and optimize procurement
decisions.

## Dataset
- Source: Kaggle
- Records: 777 purchase orders
- Format: CSV (.csv)

## Skills Used
- Python (Pandas)

In [ ]:

# PROCUREMENT KPI ANALYSIS

# Import Libraries
import pandas as pd

# Load Dataset
df = pd.read_csv("Procurement KPI Analysis Raw Data.csv")


# DATA CLEANING
# ============================================

# Convert date columns from string to datetime format
# This allows us to perform date calculations
df["Order_Date"] = pd.to_datetime(df["Order_Date"])
df["Delivery_Date"] = pd.to_datetime(df["Delivery_Date"])

# Fill missing defective units with 0
# Missing values indicate no defects were recorded
df["Defective_Units"] = df["Defective_Units"].fillna(0)

# Create new column — delivery days
# Calculates how many days each order took to arrive
df["Deliverydays"] = (df["Delivery_Date"] - df["Order_Date"]).dt.days

# Create new column — savings per order
# Calculates money saved through negotiation per order
df["Savingsperorder"] = (df["Unit_Price"] - df["Negotiated_Price"])


# ANALYSIS
# ============================================

# Question 1 — Which suppliers have highest defective units?
print("TOP 5 SUPPLIERS BY DEFECTIVE UNITS:")
print(df.groupby("Supplier")["Defective_Units"].sum().sort_values(ascending=False).head(5))

# Question 2 — Total money saved through negotiation
print("\nTOTAL NEGOTIATION SAVINGS:")
print(df["Savingsperorder"].sum())

# Question 2b — Which item category has best negotiation savings?
print("\nTOP 5 CATEGORIES BY NEGOTIATION SAVINGS:")
print(df.groupby("Item_Category")["Savingsperorder"].sum().sort_values(ascending=False).head(5))

# Question 3 — Average delivery time per supplier
print("\nAVERAGE DELIVERY DAYS PER SUPPLIER:")
print(df.groupby("Supplier")["Deliverydays"].mean().sort_values(ascending=False).head(5))

# Question 4 — Which suppliers have compliance issues?
print("\nSUPPLIER COMPLIANCE STATUS:")
print(df.groupby("Supplier")["Compliance"].value_counts())

# Question 5 — Most ordered item category
print("\nMOST ORDERED ITEM CATEGORIES:")
print(df["Item_Category"].value_counts())

TOP 5 SUPPLIERS BY DEFECTIVE UNITS:
Supplier
Delta_Logistics    19678.0
Beta_Supplies      13838.0
Gamma_Co            7034.0
Epsilon_Group       4682.0
Alpha_Inc           2717.0
Name: Defective_Units, dtype: float64

TOTAL NEGOTIATION SAVINGS:
3592.1499999999996

TOP 5 CATEGORIES BY NEGOTIATION SAVINGS:
Item_Category
MRO                806.53
Office Supplies    801.35
Packaging          673.25
Electronics        658.61
Raw Materials      652.41
Name: Savingsperorder, dtype: float64

AVERAGE DELIVERY DAYS PER SUPPLIER:
Supplier
Beta_Supplies      11.272727
Epsilon_Group      10.865772
Delta_Logistics    10.854305
Alpha_Inc          10.606838
Gamma_Co           10.192308
Name: Deliverydays, dtype: float64

SUPPLIER COMPLIANCE STATUS:
Supplier         Compliance
Alpha_Inc        Yes           132
                 No              9
Beta_Supplies    Yes           118
                 No             38
Delta_Logistics  Yes           104
                 No             67
Epsilon_Group    Y

## Data Overview
The dataset was loaded from a CSV file containing 777 purchase orders
and 13 columns. Date columns were converted to datetime format to enable
delivery time calculations. Missing defective units were filled with 0,
indicating no defects recorded. Two new columns were created — Delivery
Days and Savings Per Order — to support the analysis.

## Question 1 — Which Suppliers Have The Highest Defective Units?

**Finding:**
Through the findings Delta Logistics recorded the highest defective units
at 19,678, significantly higher than all other suppliers. This poses a
serious quality risk and warrants immediate review. While Alpha Inc had
the lowest defective units at 2,717, making them the most reliable
supplier by this metric.

## Question 2 — Negotiation Savings

**Finding A — Total Savings:**
Through negotiations across all 777 orders, the company has saved a total
of 3,592.15. This demonstrates that the procurement team's negotiation
efforts are generating measurable cost savings for the business.

**Finding B — Savings by Category:**
Through the findings MRO recorded the highest negotiation savings at 806.53,
suggesting that procurement teams are negotiating most effectively in this
category. While Raw Materials had the lowest savings at 652.41, indicating
room for stronger negotiation with Raw Materials suppliers to reduce
costs further.

## Question 3 — Average Delivery Time Per Supplier

**Finding:**
Through the findings Beta Supplies recorded the longest average delivery
time at 11.27 days, making them the slowest performing supplier and a
potential risk to operational efficiency. Gamma Co had the shortest average
delivery time at 10.19 days, making them the most reliable supplier in
terms of delivery speed.

## Question 4 — Which Suppliers Have Compliance Issues?

**Finding:**
Through the findings Gamma Co recorded the highest compliance failures
at 20 orders, making them the most non-compliant supplier despite their
fast delivery times. While Alpha Inc had fewer compliance issues at only
9 failed orders out of 141 total, demonstrating them as the most reliable
and trustworthy supplier for future procurement decisions.

## Question 5 — Most Ordered Item Category

**Finding:**
Through the findings Office Supplies was the most frequently ordered
category with 174 purchase orders, suggesting high consumption and a
consistent restocking need. Raw Materials had fewer orders at 139,
indicating lower purchasing frequency in this category.

## Overall Summary

This procurement analysis of 777 purchase orders reveals three critical
insights. First, Delta Logistics poses the greatest quality risk with the
highest defective units at 19,678 and should be reviewed urgently. Second,
Gamma Co despite being the fastest delivery supplier recorded the most
compliance failures at 20 orders, indicating that speed does not guarantee
reliability. Finally, Alpha Inc consistently emerged as the most trustworthy
supplier across both compliance and defective unit metrics, making them the
recommended supplier for future procurement decisions.